sql/core/src/test/scala/org/apache/spark/sql/StreamTest.scala

```diff
@@ -179,16 +219,40 @@ trait StreamTest extends QueryTest with Timeouts {
          |${if (streamDeathCause != null) stackTraceToString(streamDeathCause) else ""}
          |
          |== Sink ==
-         |$sink
+         |${sink.toDebugString}
          |
          |== Plan ==
          |${if (currentStream != null) currentStream.lastExecution else ""}
-         """
+         """.stripMargin
+
+    def verify(condition: => Boolean, message: String): Unit = {
+      try {
+        Assertions.assert(condition)
+      } catch {
+        case NonFatal(e) =>
+          failTest(message, e)
+      }
+    }
+
+    def eventually[T](message: String)(func: => T): T = {
+      try {
+        Eventually.eventually(Timeout(streamingTimout)) {
+          func
+        }
+      } catch {
+        case NonFatal(e) =>
+          failTest(message, e)
+      }
+    }
 
-    def checkState(check: Boolean, error: String) = if (!check) {
+    def failTest(message: String, cause: Throwable = null) = {
+      val c = Option(cause).map { e =>
+        e.getMessage + e.getStackTrace.take(10).mkString("\n\t", "\n|\t", "\n")
```

Comment:
@zsxwing I am not using Utils.exceptionToString because that does not allow the stack trace to be truncated to first few lines, and having the whole stack trace is very ugly (and not useful for debugging) in the context of all the stream state we are printing.